### PR TITLE
Get recipient keys from https:// and file:// URLs

### PR DESCRIPTION
**Pull request by paulc**
This patch includes support for https:// and file:// URLs as recipient arguments. The URL is fetched and each line is added as a recipient.

e.g.

    -r https://github.com/<user>.keys       (Use GitHub keys)
    -r file:///home/<user>/.ssh/<key>.pub   (Use local SSH key)

This is mostly useful to allow recipient keys to be specified directly from a GitHub URL (or other service which provides an equivalent)

### CODE REVIEW

**Comment by Mohammed90** [comment]
Awesome! Should resolve #34 😃

File: cmd/age/age.go [hunk 1]
```diff
@@ -169,11 +177,47 @@ func passphrasePrompt() (string, error) {
 func encryptKeys(keys []string, in io.Reader, out io.Writer, armor bool) {
 	var recipients []age.Recipient
 	for _, arg := range keys {
-		r, err := parseRecipient(arg)
+		url, err := url.Parse(arg)
```
Comment (Mohammed90) [inline cmd/age/age.go]: Rename the variable to something other than `url` so it won't clash with the package name.

**Comment by chrisiaut** [comment]
Any ETA on a merge?

**Comment by Fastidious** [comment]
I think this is a very useful PR. Any reasons why it should not be merged, @FiloSottile?

**Comment by chrisiaut** [comment]
seems like the project is dead. Development wise

**Comment by cyb3rz3us** [comment]
Not dead...just not a lot to really do.  People are always free to pull and develop as they see fit...

**Comment by Fastidious** [comment]
@cyb3rz3us something like this is mentioned on `age`'s [design document](https://docs.google.com/document/d/11yHom20CrsuX8KQJXBBw04s80Unjv8zCg_A7sPAX_9Y/preview).

**Comment by cyb3rz3us** [comment]
I wasn't arguing for or against the feature (but I do agree it is a good thing) but rather that the project is dead...it's not dead...just not on fire...

**Comment by Fastidious** [comment]
@cyb3rz3us I disregarded the "project is dead" comment altogether. I was just pointing out this PR is something that is/was wanted originally.

**Comment by cyb3rz3us** [comment]
Agreed...

**Comment by FiloSottile** [comment]
Hi! Thank you for contributing! I took a different path in #173, but it was useful to look at open PRs for the users' expectations. Let me know if that implementation works for you!

I wrote up an explanation of why I ended up implementing recipient files and not HTTPS sources on the mailing list, and I appreciate feedback! https://groups.google.com/g/age-dev/c/StmWsOyb-H8/m/a44rSBIWCAAJ

**Pull request closed without merging**